### ISSUE TITLE
Three undefined names in Python code

### ISSUE THREAD

**Issue by cclauss** [newren/git-filter-repo#128]
[flake8](http://flake8.pycqa.org) testing of https://github.com/newren/git-filter-repo on Python 3.8.3

$ __flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics__
```
./git_filter_repo.py:2804:14: F821 undefined name 'callback'
      return callback #namespace['callback']
             ^
./git_filter_repo.py:3519:51: F821 undefined name 'branch'
                         (decode(marks_basename), branch))
                                                  ^
./t/t9391/erroneous.py:12:44: F821 undefined name 'progress'
  print("Decipher this: "+''.join(reversed(progress.message)))
                                           ^
3     F821 undefined name 'callback'
3
```
https://flake8.pycqa.org/en/latest/user/error-codes.html

On the flake8 test selection, this PR does _not_ focus on "_style violations_" (the majority of flake8 error codes that [__psf/black__](https://github.com/psf/black) can autocorrect).  Instead, these tests are focus on runtime safety and correctness:
* E9 tests are about Python syntax errors usually raised because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python, they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).  Python >= 3.8 will raise SyntaxWarnings on these instances.
* F7 tests logic errors and syntax errors in type hints
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python, a __NameError__ is raised which will halt/crash the script on the user.

**Comment by newren** [comment]
I apologize for the lengthy delay in getting back to your report, especially given how high quality it is.  Thanks for taking the time to report these issues and for the pointer to flake8; it seems very cool.  Of the three cases reported:

* the last was a case meant to have an error, but somewhere else that executes earlier.  The highlighted line of code just didn't make sense and was clearly a copy-and-paste from another testcase.  While the line doesn't get executed because of an earlier error, this just makes the demos harder to understand.  I just rewrote that callback.
* the middle case was basically a typo; should have been 'full_branch' instead of branch.  It indeed was in an error case that wasn't expected to be executed, but it's nice to have the code be correct for if it ever does happen to trigger.
* the first case is a false positive.  The code is live and executed heavily by the regression tests, and the variable is very much defined, but I can see how the exec() call on the line before would be difficult for flake8 to parse.

Anyway, I fixed these all up in commit e4960a53f8 (Fix undefined variable names, 2020-07-27).
